addons/product/models/product_uom.py

```diff
@@ -0,0 +1,31 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class ProductUom(models.Model):
+    _description = 'Link between the products and the UoM'
+
+    uom_id = fields.Many2one('uom.uom', 'Unit of Measure', required=True, ondelete='cascade')
+    product_id = fields.Many2one('product.product', 'Product', required=True, ondelete='cascade')
+    name = fields.Char('Barcode', index='btree_not_null', required=True)
```

Comment:
This is messing with my brain :dizzy_face:

Can you rename this field to `barcode`? I don't think we need a `name` field since you override the display name compute anyways?